interactions/client/bot.py

```diff
@@ -708,22 +728,27 @@ def __check_options(_option: Option, _names: list, _sub_command: Option = MISSIN
         def __check_coro():
             __indent = 4
             log.debug(f"{' ' * __indent}Checking coroutine: '{coro.__name__}'")
-            if not len(coro.__code__.co_varnames):
+            _isMethod = hasattr(coro, "__func__")
+            if not len(coro.__code__.co_varnames) ^ (
+                _isMethod and len(coro.__code__.co_varnames) == 1
+            ):
                 raise InteractionException(
                     11, message="Your command needs at least one argument to return context."
                 )
             elif "kwargs" in coro.__code__.co_varnames:
                 return
-            elif _sub_cmds_present and len(coro.__code__.co_varnames) < 2:
+            elif _sub_cmds_present and len(coro.__code__.co_varnames) < (3 if _isMethod else 2):
                 raise InteractionException(
                     11, message="Your command needs one argument for the sub_command."
                 )
-            elif _sub_groups_present and len(coro.__code__.co_varnames) < 3:
+            elif _sub_groups_present and len(coro.__code__.co_varnames) < (4 if _isMethod else 3):
                 raise InteractionException(
                     11,
                     message="Your command needs one argument for the sub_command and one for the sub_command_group.",
                 )
-            add: int = 1 + abs(_sub_cmds_present) + abs(_sub_groups_present)
+            add: int = (
+                1 + abs(_sub_cmds_present) + abs(_sub_groups_present) + 1 if _isMethod else +0
+            )
```

Comment:
Why camelCase?
Also why `+0`????
```suggestion
            _is_method = hasattr(coro, "__func__")
            if not len(coro.__code__.co_varnames) ^ (
                _is_method and len(coro.__code__.co_varnames) == 1
            ):
                raise InteractionException(
                    11, message="Your command needs at least one argument to return context."
                )
            elif "kwargs" in coro.__code__.co_varnames:
                return
            elif _sub_cmds_present and len(coro.__code__.co_varnames) < (3 if _is_method else 2):
                raise InteractionException(
                    11, message="Your command needs one argument for the sub_command."
                )
            elif _sub_groups_present and len(coro.__code__.co_varnames) < (4 if _is_method else 3):
                raise InteractionException(
                    11,
                    message="Your command needs one argument for the sub_command and one for the sub_command_group.",
                )
            add: int = (
                1 + abs(_sub_cmds_present) + abs(_sub_groups_present) + 1 if _is_method else 0
            )
```

Reply:
camel case will be resolved, 
![grafik](https://user-images.githubusercontent.com/88881326/167082694-3be139d2-dd68-45ab-91ca-a28217223787.png)